jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.c

```diff
@@ -920,6 +921,56 @@ ecma_builtin_string_prototype_object_slice (ecma_string_t *get_string_val, /**<
   return ecma_make_string_value (new_str_p);
 } /* ecma_builtin_string_prototype_object_slice */
 
+/**
+ * The String.prototype object's 'at' routine
+ *
+ * See also:
+ *          ECMA-262 Stage 3 Draft Relative Indexing Method proposal
+ *          from: https://tc39.es/proposal-relative-indexing-method
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_string_prototype_object_at (ecma_string_t *string_val, /**< this argument */
+                                         const ecma_value_t index) /**< index argument */
+{
+
+  const lit_utf8_size_t len = ecma_string_get_length (string_val);
+
+  ecma_number_t relative_index;
```

Comment:
This part repeats several times. Maybe a helper function could be introduced for this.